File: sources/springboot-api/src/main/java/com/example/springbootapi/MongoService.java
package com.example.springbootapi;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.MongoCollection;
import org.bson.Document;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.ArrayList;

@Service
public class MongoService {
    private final String mongoUri = System.getenv().getOrDefault("SPRING_DATA_MONGODB_URI", "mongodb://localhost:27017/shared_db");
    private final String dbName = "shared_db";

    public boolean ping() {
        try (MongoClient mongoClient = MongoClients.create(mongoUri)) {
            MongoDatabase db = mongoClient.getDatabase(dbName);
            db.runCommand(new Document("ping", 1));
            return true;
        } catch (Exception e) {
            return false;
        }
    }

    public List<Document> findDocuments(String collection, Document query) {
        try (MongoClient mongoClient = MongoClients.create(mongoUri)) {
            MongoDatabase db = mongoClient.getDatabase(dbName);
            MongoCollection<Document> coll = db.getCollection(collection);
            return coll.find(query).into(new ArrayList<>());
        } catch (Exception e) {
            return new ArrayList<>();
        }
    }
}

File: sources/nestjs-api/src/mongo.service.ts
import { Injectable } from '@nestjs/common';
import { MongoClient } from 'mongodb';

@Injectable()
export class MongoService {
  private readonly mongoUrl = process.env.MONGO_URL || 'mongodb://mongo:27017/shared_db';

  async ping(): Promise<boolean> {
    let client: MongoClient | null = null;
    try {
      client = new MongoClient(this.mongoUrl);
      await client.db().admin().ping();
      await client.close();
      return true;
    } catch {
      if (client) await client.close();
      return false;
    }
  }

  async findDocuments(collection: string, query = {}): Promise<any[]> {
    let client: MongoClient | null = null;
    try {
      client = new MongoClient(this.mongoUrl);
      await client.connect();
      const db = client.db();
      const docs = await db.collection(collection).find(query).toArray();
      return docs;
    } catch (e) {
      return [];
    } finally {
      if (client) await client.close();
    }
  }
}
